src/long.ts

```diff
@@ -406,7 +406,11 @@ export class Long extends BSONValue {
       unsigned = !!unsignedOrRadix;
     }
     radix ??= 10;
-    if (str === 'NaN' || str === 'Infinity' || str === '+Infinity' || str === '-Infinity') {
+    if (str === 'NaN' && radix < 24) {
+      // radix does not support n, so coerce to zero
+      return Long.ZERO;
+    } else if ((str === 'Infinity' || str === '+Infinity' || str === '-Infinity') && radix < 35) {
+      // radix does not support y, so coerce to zero
```

Comment:
This doesn't account for lower case or mixed case inputs. Is this intended to be undefined behaviour? If so, we should document that somewhere, otherwise we should decide on an approach to standardize this behaviour.

```
> Long.fromString('infinity', 25) // new Long('185011843')

> Long.fromString("InFinity", 25) // new Long('185011843')

> Long.fromString("Infinity", 25) // new Long('0')

```

Reply:
This is the desired behavior. If a user inputs non-valid characters in a radix, they might get undefined behavior. In the case that they input the actual js syntax for `Infinity` the driver will coerce the input into something manageable.

Since this PR intends to fix the output for valid numeric inputs, not coercions for invalid inputs, this wasn't something I considered in scope.

Reply:
I added in some more documentation to fromString though!